### PR TITLE
Test practitioner form

### LINKED ISSUE #526: TeamAssignment - Add Tests for Practitioner Form
- [ ] add tests  to the practitioner form component
- [ ] add tests for practitionerform.userIdSelect component

### PR DESCRIPTION
**Pull request by p-netm**
Adds tests for this components:
- [x] Practitioner form
- [x] UserIdSelect

closes #526

### CODE REVIEW

File: src/containers/forms/PractitionerForm/UserIdSelect/index.tsx [hunk 1]
```diff
@@ -50,30 +51,26 @@ export const UserIdSelect: React.FC<Props> = props => {
...
-      responseSize = userData.results.length;
+      // modify filter params to point to next page
+      const responseSize = response.results.length;
       filterParams = {
         ...filterParams,
         start_index: filterParams.start_index + responseSize,
       };
-      allOpenMRSUsers.push(...userData.results);
-    }
+    } while (thereIsNextPage(response));
```
Comment (cKellyDesign) [inline src/containers/forms/PractitionerForm/UserIdSelect/index.tsx]: Nice `do` block! Are we catching potential errors in this loop?

File: src/containers/forms/PractitionerForm/UserIdSelect/index.tsx [hunk 2]
```diff
@@ -120,3 +117,26 @@ export const UserIdSelect: React.FC<Props> = props => {
...
 UserIdSelect.defaultProps = defaultProps;
 export default UserIdSelect;
+
+/** interface for paginated response got from openSRP
+ * openMRS proxy for getting openMRS users.
+ */
+interface OpenMRSResponse {
+  links: Array<{ rel: string; uri: string }>;
+  results: OpenMRSUser[];
+}
```
Comment (cKellyDesign) [inline src/containers/forms/PractitionerForm/UserIdSelect/index.tsx]: Should we `export` this?

File: src/containers/forms/PractitionerForm/UserIdSelect/tests/index.test.tsx [hunk 3]
```diff
@@ -36,6 +38,42 @@ describe('src/*/forms/userIdSelect', () => {
...
+  it('calls to fetch', async () => {
+    fetch.once(JSON.stringify(practitioners)).once(JSON.stringify(openMRSUsers));
+
+    const props = {
+      serviceClass: OpenSRPService,
+    };
+    mount(<UserIdSelect {...props} />);
+    await new Promise(resolve => new Promise(resolve));
+    await flushPromises();
+    const calls = [
```
Comment (cKellyDesign) [inline src/containers/forms/PractitionerForm/UserIdSelect/tests/index.test.tsx]: Nice, these are great 👏

File: src/containers/forms/PractitionerForm/UserIdSelect/index.tsx (hunk 2 above)
Comment (p-netm) [reply src/containers/forms/PractitionerForm/UserIdSelect/index.tsx]: exported

**Pull request merged**